sortableidentifier/src/main/java/com/github/awsjavakit/identifiers/SortableIdentifier.java

```diff
@@ -0,0 +1,45 @@
+package com.github.awsjavakit.identifiers;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import java.time.Instant;
+import java.util.UUID;
+
+public class SortableIdentifier {
+
+  private static final String CONSTANT_TIMESTAMP_LENGTH = "%012x-%s";
+  private static final int UUID_START_INDEX = 13;
+  private final String value;
+
+  @JsonCreator
+  public SortableIdentifier(String value) {
+    validate(value);
```

Comment:
I'm always tempted to say that validators should be configurable and external, but I guess that what a SortableIdentifier is is and only ever will be this thing

Reply:
The validation should be as cheap as possible because it is always invoked upon creation. We can refactor later if we need to parameterize